ostelco-ios-client/ViewControllers/EKYC/PendingVerificationViewController.swift

```diff
@@ -19,25 +19,68 @@ class PendingVerificationViewController: UIViewController {
     }
     
     @IBAction private func `continue`(_ sender: Any) {
+        checkVerificationStatus()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        addNotificationObserver(selector: #selector(onDidReceiveData(_:)))
+        addWillEnterForegroundObserver(selector: #selector(didBecomeActive))
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        removeNotificationObserver()
+        removeWillEnterForegroundObserver()
+    }
+    
+    @objc func onDidReceiveData(_ notification: Notification) {
+        print(#function, "Notification didReceivePushNotification arrived")
+    }
+    
+    @objc func didBecomeActive() {
+        checkVerificationStatus(silentCheck: true)
+    }
+    
+    func checkVerificationStatus(silentCheck: Bool = false) {
         let countryCode = OnBoardingManager.sharedInstance.selectedCountry.countryCode.lowercased()
         let spinnerView = showSpinner(onView: self.view)
         APIManager.sharedInstance.regions.child(countryCode).load()
             .onSuccess { data in
                 if let regionResponse: RegionResponse = data.typedContent(ifNone: nil) {
                     // if let regionRespons.kycStatusMap.NR
                     // TODO: Convert status to enum
-                    if regionResponse.status == .APPROVED {
+                    switch regionResponse.status {
+                    case .APPROVED:
                         self.performSegue(withIdentifier: "ESim", sender: self)
-                    } else {
-                        if let status = regionResponse.kycStatusMap.JUMIO {
-                            self.showAlert(title: "Status", msg: status.rawValue)
-                        } else {
+                    case .PENDING:
+                        if !silentCheck {
                             self.showAlert(title: "Status", msg: regionResponse.status.rawValue)
                         }
+                    case .REJECTED:
+                        if let jumioStatus = regionResponse.kycStatusMap.JUMIO, let nricStatus = regionResponse.kycStatusMap.NRIC_FIN, let addressStatus = regionResponse.kycStatusMap.ADDRESS_AND_PHONE_NUMBER {
```

Comment:
worth pulling this handling out into a separate method since it's got its own nested switch statement?

Reply:
Yeah, could do that for each case on top level at least

Reply:
That might even be overkill for the simpler stuff, but this one starts getting into quite a bit more logic, so it seems a bit more natural to pull out